### ISSUE TITLE
different node has different parameters

### ISSUE THREAD

**Issue by YihengJiang** [pytorch/examples#483]
I have tried it, but if I found that each model in all node has different gradients,so it results to different model among GPUs, At last I do like this:
            #something to do###############
            loss.backward()
            self.average_gradients()
            self.optimizer.step():
           #other thing to do###############

    def average_gradients(self):
        world_size = distributed.get_world_size()

        for p in self.net.parameters():
            distributed.all_reduce(p.grad.data, op=distributed.reduce_op.SUM)
            p.grad.data /= float(world_size)

It work normally,but I do not know whether it is right, cause official of pyTorch do not mention it.
could you tell me is it right? thank you!!!

And another question: I found I can not run on 2 or more machines, I do not know how to configure it, should I make a configur so that all machines in my group can access each other without password by ssh?

**Comment by soumith** [comment]
Hi, your question does not have any context.
I am not sure what you are asking about, but if you want to all-reduce the gradients, then yes this line should work.

Please use https://discuss.pytorch.org for questions, and any further follow-up
